specutils/spectra/spectral_coordinate.py

```diff
@@ -676,7 +685,9 @@ def with_los_shift(self, target_shift=None, observer_shift=None):
         init_obs_vel = self._calculate_radial_velocity(observer_icrs, target_icrs)
         fin_obs_vel = self._calculate_radial_velocity(new_observer, new_target)
 
-        new_data = self._project_velocity_and_shift(init_obs_vel, fin_obs_vel)
+        line_of_sight_unit_vec = self._norm_d_pos(observer_icrs, target_icrs).xyz
```

Comment:
I was a little puzzled here that I don't get the same result if I use ``new_observer`` and ``new_target`` - @eteq, is that because ``_norm_d_pos`` assumes the coordinates are in the same frame?

Reply:
Yes that's right.  And it probably shouldn't do that - I'd suggest adding this to the top of `_norm_d_pos`:
```
target_in_obs_frame = target.transform_to(observer)
```
and then use `target_in_obs_frame` everywhere that `target` is currently used.  That should be a no-op if they're already in the same frame, but will protect against mis-use like it is right now.
```